remotespark/livyclientlib/livysession.py

```diff
@@ -68,13 +68,21 @@ def create_sql_context(self):
 
         self.logger.debug("Starting '{}' sql and hive session.".format(self.language))
 
-        self.wait_for_status(Constants.idle_session_status, self._create_sql_context_timeout_seconds)
-        self.execute(self._get_sql_context_creation_command())
-        self.logger.debug("Started '{}' sql session.".format(self.language))
-
-        self.wait_for_status(Constants.idle_session_status, self._create_sql_context_timeout_seconds)
-        self.execute(self._get_hive_context_creation_command())
-        self.logger.debug("Started '{}' hive session.".format(self.language))
+        try:
+            self.wait_for_status(Constants.idle_session_status, self._create_sql_context_timeout_seconds)
```

Comment:
Will this method `wait_for_status` ever be called with a session status that is not `idle`? If so then you can consider changing the method to `wait_for_idle` and just doing away with the first argument